src/selectors/getSelectedFrameLocation.js

```diff
@@ -0,0 +1,21 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */
+
+import { getSelectedLocation } from "../reducers/sources";
+import { getSelectedFrame } from "../reducers/pause";
+import { isOriginalId } from "devtools-source-map";
+
+export default function getSelectedFrameLocation(state: OuterState) {
+  const selectedLocation = getSelectedLocation(state);
+  const isGeneratedSource = !isOriginalId(selectedLocation.sourceId);
+  const selectedFrame = getSelectedFrame(state);
+
+  if (!selectedFrame) {
+    return;
+  }
+
+  return isGeneratedSource
+    ? selectedFrame.generatedLocation || selectedFrame.location
+    : selectedFrame.location;
+}
```

Comment:
I prefer what we do w/ getVisibleBreakpoints and returning the entire object for a couple of reasons

- the debug line still needs to know if we're paused on an error
- the debug line could need the frame ID...
- seems more future proof

Reply:
Cool. beside location, which other props from the frame would you like to copy?
right now it's the only one that is really needed in DebugLine.js